### ISSUE TITLE
DEV:  ShareCodeUrl for room doesn't work on Android

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#24505]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any account, create a new room (make sure you have your local web running.)
2. Go to that room detail => Share code => Click Copy url to clipboard, paste it to your browser and press Enter.
3. Notice that the browser can't load your url.
## Expected Result: 
Share code url should work with local environment on Android
## Actual Result: 
Share Code Url doesn't work on Android.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 
**Reproducible in staging?:** Dev
**Reproducible in production?:**  Dev
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/c0d5ee29-4ca5-46c7-8104-4a835cda8ab3

https://github.com/Expensify/App/assets/43996225/fa2b744a-1b4f-428f-a307-d7225c41949c

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hungvu193 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1691050736969549

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kavimuru** [comment]
## Proposal by @hungvu193 

### Please re-state the problem that we are trying to solve in this issue.
DEV:  ShareCodeUrl for room doesn't work on Android.
### What is the root cause of that problem?
As we can see the share code url in this issue still had the old port: 8080.
We decided to change the port to `8082` few weeks ago to make Flipper work on mobile, but we forgot also update the default value for port in [here](https://github.com/Expensify/App/blob/157abce51a8ba4c1b7a82a596dcc6686b82be76d/src/CONFIG.js#L83) and [here](https://github.com/Expensify/App/blob/157abce51a8ba4c1b7a82a596dcc6686b82be76d/desktop/main.js#L14) which caused the issue on mobile.
### What changes do you think we should make in order to solve the problem?
We should update the default value of the port to `8082` in [here](https://github.com/Expensify/App/blob/157abce51a8ba4c1b7a82a596dcc6686b82be76d/src/CONFIG.js#L83) and [here](https://github.com/Expensify/App/blob/157abce51a8ba4c1b7a82a596dcc6686b82be76d/desktop/main.js#L14)
```javascript
DEV_PORT: process.env.PORT || 8082

// and

const port = process.env.PORT || 8082

```

We can also add the `PORT=8082` to our .env files and use the value from Config.PORT as a default value.


### What alternative solutions did you explore? (Optional)
N/A

**Comment by garrettmknight** [comment]
Asking an engineer to reproduce on dev: https://expensify.slack.com/archives/C03TQ48KC/p1692025561675429

**Comment by cristipaval** [comment]
yes the port is still hardcoded in those 2 places and the issue is reproducible
cc'ing @amyevans as I think she worked on changing the port for dev environment.

**Comment by garrettmknight** [comment]
Thanks for the help @cristipaval

**Comment by amyevans** [comment]
Agreed with Cristi. It's a straightforward fix, we can open it up to external to knock it out, or I can take it as a Weekly to handle in some downtime, either way!

**Comment by burczu** [comment]
@garrettmknight yes, I agree - we can go with @kavimuru fix here.

🎀 👀 🎀 C+ reviewed

**Comment by burczu** [comment]
@hungvu193 Ahhh, right! I've just noticed it - sorry for my mistake - updated the comment ;)

**Comment by thienlnam** [comment]
@burczu Could you please remember to link to the proposal in future approved comments? Thank you!

**Comment by burczu** [comment]
@thienlnam sure thing!

**Comment by garrettmknight** [comment]
Looks like this one didn't get triggered for payment - confirmed that the bug is fixed, I'll process payment now.

**Comment by garrettmknight** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@burczu] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@burczu] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@burczu] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@burczu] Determine if we should create a regression test for this bug.
- [ ] [@burczu] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
- [ ]  [@garrettmknight] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by garrettmknight** [comment]
**Summary of payments for this issue:**

- #urgency bonus? Yes
- Reporter: @hungvu193 - $250 paid via Upwork
- Contirbutor: @hungvu193 - $1500 paid via Upwork
- Contributor+: @burczu - $1500

Upwork job: https://www.upwork.com/jobs/~0174a2b1b05a63292e

**Comment by garrettmknight** [comment]
@burczu can you go through the checklist when you get a chance?

**Comment by burczu** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@burczu] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/21243
* [x]  [@burczu] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/21243#issuecomment-1708033010
* [x]  [@burczu] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not found
* [x]  [@burczu] Determine if we should create a regression test for this bug.
* [ ]  [@burczu] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
* [ ]   [@garrettmknight] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by burczu** [comment]
@garrettmknight Regarding this comment: https://github.com/Expensify/App/issues/24505#issuecomment-1705003473 - I won't be paid because I'm from Callstack, an experts contributor group.

**Comment by garrettmknight** [comment]
Oops, good catch! Closing this one out!
